app/forms/disk-create.tsx

```diff
@@ -79,6 +80,19 @@ export function CreateDiskSideModalForm({
   })
 
   const form = useForm({ defaultValues })
+  const control = form.control
+  const { project } = useProjectSelector()
+  const projectImages = useApiQuery('imageList', { query: { project } })
+  const siloImages = useApiQuery('imageList', {})
+
+  // put project images first because if there are any, there probably aren't
+  // very many and they're probably relevant
+  const images = [...(projectImages.data?.items || []), ...(siloImages.data?.items || [])]
+  const areImagesLoading = projectImages.isPending || siloImages.isPending
+
+  const selectedImageId = control?._formValues?.diskSource?.imageId
```

Comment:
```suggestion
  const selectedImageId = form.watch('diskSource.imageId')
```

This ensures the component re-renders when the image changes. If you `console.log(selectedImage)` without the `watch`, you can see that it does not render when it should.

https://react-hook-form.com/docs/useform/watch

Reply:
`form.watch` is so much more graceful for snagging that data